mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -389,8 +378,16 @@ LAGenericRewritePattern::matchAndRewrite(linalg::GenericOp laGeneric,
   // 2.2.0. Move the generic before the write-back. This'll put all
   // the copy loops for other inputs before the generic due to insertion
   // order.
-  gemmStoreOp->moveAfter(laGeneric);
 
+  // Before being copy-optimized the output of prior op (could be gemm-like rock
+  // op or linalg block) will be written to the internal memref alloca. If that
+  // is not copied as an output, it is not an actual output. Therefore, we dont
+  // need clone the gemmStore.
+  if (isLeadingToBlockArg(gemmStoreOp.getDest().getDefiningOp())) {
```

Comment:
Ok, I see what's going on here, now that you've restructured things. Can I ask for a different name for this function? Perhaps `leadsToUnwrittenBlockArg`

Reply:
done!